nixos/modules/programs/_1password-gui.nix

```
@@ -0,0 +1,55 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.programs._1password-gui;
+
+in {
+  options = {
+    programs._1password-gui = {
+      enable = mkEnableOption "The 1Password Desktop application with browser integration";
+      groupId = mkOption {
+        default = 44399;
+        defaultText = "44399";
+        example = "44399";
+        description = "This provides an ID for the onepassword group, which will own the setgid helper applications. This value can be anything, so long as it is unique on the system. 44399 was merely chosen randomly during development.";
+        type = types.ints.positive;
+      };
+    };
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs._1password-gui;
+      defaultText = literalExpression "pkgs._1password-gui";
+      example = literalExpression "pkgs._1password-gui";
+      description = ''
+        The 1Password derivation to use. This can be used to upgrade from the stable release that we keep in nixpkgs to the betas.
+        '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    environment.systemPackages = [ package ];
+    users.groups.onepassword.gid = cfg.groupId;
```

```suggestion
    users.groups.onepassword = { };
```
since you say ID was chosen randomly. we *must not* chose random IDs like this without listing them in `ids.nix`, but if the actual numeric ID is not important we may as well have the system assign a unique ID when needed. this also eliminates the `groupId` option.

---

Switched. Thanks!

---

@pennae I've discovered that we can't be *completely* random in the `gid` choice. Basically, 1Password requires that the `gid` be >= 1000. Is there a way for me to restrict that? Alternatively, would the team be open to me adding a new mechanism for restricting the `gid`, similar to `subGidRanges` that apply to users?

---

that does sound ... rather problematic :confused: a static "large" gid could interfere with mutableUsers = false, so that's out immediately. adding a valid range for each gid *should* be possible but could also have unfortunate effects unless the range given are excessively large. adding that option should certainly be possible, but no idea whether it would be acceptable to others. feels like the better solution would be to somehow get 1password to accept gids < 1000 as well, but no clue what that would take :(

---

If it's difficult / impossible to get 1Password to allow gids < 1000, it is possible to specify a static guid here: https://github.com/NixOS/nixpkgs/blob/644ad81cc54d4741cd0b26336d6ce1418ca7afa3/nixos/modules/misc/ids.nix#L364

And then I'd just use that as the gid, via `gid = config.ids.gid.onepassword`.